src/data/batch_buffer.py

```
@@ -0,0 +1,86 @@
+        """
+        Fill buffer with token IDs.
+        
+        Args:
+            token_ids: List of token IDs to fill buffer with
+            actual_length: Actual length for attention mask (if different from token count)
+        """
+        num_tokens = len(token_ids)
+        mask_length = actual_length if actual_length is not None else num_tokens
+        
+        if num_tokens > self.max_seq_length:
+            raise ValueError(f"Token count {num_tokens} exceeds max_seq_length {self.max_seq_length}")
```

Ruff (0.14.6)

Avoid specifying long messages outside the exception class

(TRY003)